File: n8n-tests/test-all-webhooks.py
#!/usr/bin/env python3
"""
Globul Cars - n8n Webhooks Automated Testing Script (Python)

Tests all 21 webhook endpoints automatically
Run: python test-all-webhooks.py
"""

import requests
import json
import time
from datetime import datetime
from typing import Dict, List, Tuple

BASE_URL = 'https://globul-cars-bg.app.n8n.cloud/webhook'

# Color codes for terminal output
class Colors:
    GREEN = '\033[92m'
    RED = '\033[91m'
    YELLOW = '\033[93m'
    BLUE = '\033[94m'
    CYAN = '\033[96m'
    RESET = '\033[0m'

def log_success(msg: str):
    print(f"{Colors.GREEN}✓{Colors.RESET} {msg}")

def log_error(msg: str):
    print(f"{Colors.RED}✗{Colors.RESET} {msg}")

def log_info(msg: str):
    print(f"{Colors.BLUE}ℹ{Colors.RESET} {msg}")

def log_warn(msg: str):
    print(f"{Colors.YELLOW}⚠{Colors.RESET} {msg}")

def log_section(msg: str):
    print(f"\n{Colors.CYAN}▶{Colors.RESET} {msg}")

def get_timestamp() -> str:
    return datetime.utcnow().isoformat() + 'Z'

def get_unique_id(prefix: str = 'test') -> str:
    return f"{prefix}-{int(time.time() * 1000)}"

# Test data for all endpoints
WEBHOOK_TESTS = [
    # Workflow 1: Sell Process Started
    {
        'workflow': 'Sell Process',
        'endpoint': '/sell-started',
        'body': {
            'userId': get_unique_id('user'),
            'timestamp': get_timestamp(),
            'source': 'automated_test_python',
            'userProfile': {
                'displayName': 'Python Test User',
                'email': 'test@globul.net',
                'language': 'bg'
            }
        }
    },

    # Workflow 2: Vehicle Type Selected
    {
        'workflow': 'Vehicle Type',
        'endpoint': '/vehicle-type-selected',
        'body': {
            'userId': get_unique_id('user'),
            'vehicleType': 'car',
            'timestamp': get_timestamp(),
            'sessionId': get_unique_id('session')
        }
    },

    # Workflow 3: Complete Sell Workflow
    {
        'workflow': 'Complete Sell',
        'endpoint': '/seller-type-selected',
        'body': {
            'userId': get_unique_id('user'),
            'sellerType': 'private',
            'timestamp': get_timestamp(),
            'sessionId': get_unique_id('session')
        }
    },
    {
        'workflow': 'Complete Sell',
        'endpoint': '/vehicle-data-entered',
        'body': {
            'userId': get_unique_id('user'),
            'sessionId': get_unique_id('session'),
            'vehicleData': {
                'make': 'BMW',
                'model': 'X5',
                'year': 2020,
                'mileage': 45000,
                'fuelType': 'diesel',
                'transmission': 'automatic'
            },
            'timestamp': get_timestamp()
        }
    },
    {
        'workflow': 'Complete Sell',
        'endpoint': '/equipment-selected',
        'body': {
            'userId': get_unique_id('user'),
            'sessionId': get_unique_id('session'),
            'equipment': ['air_conditioning', 'navigation', 'leather_seats'],
            'timestamp': get_timestamp()
        }
    },
    {
        'workflow': 'Complete Sell',
        'endpoint': '/images-uploaded',
        'body': {
            'userId': get_unique_id('user'),
            'sessionId': get_unique_id('session'),
            'images': [
                {'url': 'https://example.com/car1.jpg', 'type': 'exterior', 'order': 1},
                {'url': 'https://example.com/car2.jpg', 'type': 'interior', 'order': 2}
            ],
            'totalImages': 2,
            'timestamp': get_timestamp()
        }
    },
    {
        'workflow': 'Complete Sell',
        'endpoint': '/price-set',
        'body': {
            'userId': get_unique_id('user'),
            'sessionId': get_unique_id('session'),
            'price': 35000,
            'currency': 'EUR',
            'negotiable': True,
            'timestamp': get_timestamp()
        }
    },
    {
        'workflow': 'Complete Sell',
        'endpoint': '/contact-info-entered',
        'body': {
            'userId': get_unique_id('user'),
            'sessionId': get_unique_id('session'),
            'contactInfo': {
                'name': 'Python Test',
                'phone': '+359888123456',
                'email': 'test@globul.net',
                'preferredContact': 'phone'
            },
            'timestamp': get_timestamp()
        }
    },
    {
        'workflow': 'Complete Sell',
        'endpoint': '/listing-published',
        'body': {
            'userId': get_unique_id('user'),
            'sessionId': get_unique_id('session'),
            'listingId': get_unique_id('listing'),
            'status': 'published',
            'publishedAt': get_timestamp()
        }
    },

    # Workflow 4: User Tracking
    {
        'workflow': 'User Tracking',
        'endpoint': '/user-registered',
        'body': {
            'userId': get_unique_id('user'),
            'email': 'test@globul.net',
            'displayName': 'Python Test User',
            'registrationMethod': 'email',
            'timestamp': get_timestamp()
        }
    },
    {
        'workflow': 'User Tracking',
        'endpoint': '/user-logged-in',
        'body': {
            'userId': get_unique_id('user'),
            'timestamp': get_timestamp(),
            'loginMethod': 'email',
            'deviceInfo': {
                'type': 'desktop',
                'os': 'Windows',
                'browser': 'Python Script'
            }
        }
    },
    {
        'workflow': 'User Tracking',
        'endpoint': '/profile-updated',
        'body': {
            'userId': get_unique_id('user'),
            'timestamp': get_timestamp(),
            'updatedFields': ['phone', 'location']
        }
    },
    {
        'workflow': 'User Tracking',
        'endpoint': '/listing-created',
        'body': {
            'userId': get_unique_id('user'),
            'listingId': get_unique_id('listing'),
            'vehicleType': 'car',
            'make': 'BMW',
            'model': 'X5',
            'year': 2020,
            'price': 35000,
            'timestamp': get_timestamp()
        }
    },

    # Workflow 5: User Engagement
    {
        'workflow': 'User Engagement',
        'endpoint': '/car-viewed',
        'body': {
            'userId': get_unique_id('user'),
            'carId': get_unique_id('car'),
            'timestamp': get_timestamp(),
            'viewDuration': 45,
            'source': 'search_results'
        }
    },
    {
        'workflow': 'User Engagement',
        'endpoint': '/favorite-added',
        'body': {
            'userId': get_unique_id('user'),
            'carId': get_unique_id('car'),
            'timestamp': get_timestamp(),
            'source': 'listing_page'
        }
    },
    {
        'workflow': 'User Engagement',
        'endpoint': '/message-sent',
        'body': {
            'event': 'message_sent',
            'messageId': get_unique_id('msg'),
            'chatId': get_unique_id('chat'),
            'carId': get_unique_id('car'),
            'senderId': get_unique_id('user'),
            'receiverId': get_unique_id('seller'),
            'text': 'Test message from Python script',
            'language': 'en',
            'timestamp': get_timestamp()
        }
    },
    {
        'workflow': 'User Engagement',
        'endpoint': '/search-performed',
        'body': {
            'userId': get_unique_id('user'),
            'timestamp': get_timestamp(),
            'searchQuery': {
                'make': 'BMW',
                'yearFrom': 2018,
                'yearTo': 2023
            },
            'resultsCount': 15
        }
    },

    # Workflow 6: Admin & BI
    {
        'workflow': 'Admin & BI',
        'endpoint': '/admin-login',
        'body': {
            'action': 'admin_login',
            'adminId': 'admin-test',
            'timestamp': get_timestamp(),
            'data': {
                'isNewLocation': False,
                'isNewDevice': False,
                'multipleFailedAttempts': False
            }
        }
    },
    {
        'workflow': 'Admin & BI',
        'endpoint': '/admin-action',
        'body': {
            'action': 'admin_action',
            'adminId': 'admin-test',
            'timestamp': get_timestamp(),
            'data': {
                'operation': 'view_dashboard',
                'targetId': 'dashboard-main'
            }
        }
    },
    {
        'workflow': 'Admin & BI',
        'endpoint': '/analytics-viewed',
        'body': {
            'action': 'analytics_viewed',
            'adminId': 'admin-test',
            'timestamp': get_timestamp(),
            'data': {
                'view': 'sales_dashboard',
                'filters': {'period': '7d'}
            }
        }
    },
    {
        'workflow': 'Admin & BI',
        'endpoint': '/system-alert',
        'body': {
            'action': 'system_alert',
            'adminId': 'admin-test',
            'timestamp': get_timestamp(),
            'data': {
                'severity': 'low',
                'alertType': 'test_alert',
                'message': 'Automated test from Python'
            }
        }
    }
]

def test_webhook(test: Dict) -> Tuple[bool, int, str]:
    """Test a single webhook endpoint"""
    url = BASE_URL + test['endpoint']
    
    try:
        response = requests.post(
            url,
            json=test['body'],
            headers={'Content-Type': 'application/json'},
            timeout=10
        )
        
        status = response.status_code
        success = 200 <= status < 300
        
        if success:
            log_success(f"[{test['workflow']}] {test['endpoint']} → {status}")
        else:
            log_error(f"[{test['workflow']}] {test['endpoint']} → {status}")
            log_warn(f"   Response: {response.text[:100]}")
        
        return success, status, response.text
        
    except Exception as e:
        log_error(f"[{test['workflow']}] {test['endpoint']} → FAILED")
        log_warn(f"   Error: {str(e)}")
        return False, 0, str(e)

def run_all_tests():
    """Run all webhook tests"""
    print('\n╔═══════════════════════════════════════════════════════╗')
    print('║  Globul Cars - n8n Webhooks Automated Test Suite     ║')
    print('╚═══════════════════════════════════════════════════════╝\n')
    
    log_info(f"Testing {len(WEBHOOK_TESTS)} endpoints...")
    log_info(f"Base URL: {BASE_URL}\n")
    
    results = []
    current_workflow = ''
    
    for test in WEBHOOK_TESTS:
        if test['workflow'] != current_workflow:
            log_section(f"Testing {test['workflow']}")
            current_workflow = test['workflow']
        
        success, status, response = test_webhook(test)
        results.append({
            'test': test,
            'success': success,
            'status': status,
            'response': response
        })
        
        # Small delay between requests
        time.sleep(0.5)
    
    # Summary
    print('\n╔═══════════════════════════════════════════════════════╗')
    print('║  Test Summary                                         ║')
    print('╚═══════════════════════════════════════════════════════╝\n')
    
    successful = sum(1 for r in results if r['success'])
    failed = len(results) - successful
    total = len(results)
    success_rate = (successful / total * 100) if total > 0 else 0
    
    log_info(f"Total Tests: {total}")
    log_success(f"Successful: {successful}")
    if failed > 0:
        log_error(f"Failed: {failed}")
    log_info(f"Success Rate: {success_rate:.1f}%")
    
    if failed > 0:
        print(f"\n{Colors.YELLOW}Failed Endpoints:{Colors.RESET}")
        for r in results:
            if not r['success']:
                endpoint = r['test']['endpoint']
                status = r['status'] or 'Network Error'
                print(f"  - {endpoint} ({status})")
    
    print(f"\n{Colors.CYAN}Next Steps:{Colors.RESET}")
    print("  1. Check n8n Executions: https://globul-cars-bg.app.n8n.cloud")
    print("  2. Verify workflows are Active")
    print("  3. Review any failed endpoints above\n")
    
    return 0 if failed == 0 else 1

if __name__ == '__main__':
    try:
        exit_code = run_all_tests()
        exit(exit_code)
    except KeyboardInterrupt:
        print(f"\n{Colors.YELLOW}Tests interrupted by user{Colors.RESET}")
        exit(1)
    except Exception as e:
        log_error(f"Fatal error: {str(e)}")
        exit(1)

File: DDD/UNNECESSARY_FILES_OCT_22/n8n-tests/test-all-webhooks.js
#!/usr/bin/env node

/**
 * Globul Cars - n8n Webhooks Automated Testing Script
 * 
 * Tests all 21 webhook endpoints automatically
 * Run: node test-all-webhooks.js
 */

const BASE_URL = 'https://globul-cars-bg.app.n8n.cloud/webhook';

// Color output helpers
const colors = {
  reset: '\x1b[0m',
  green: '\x1b[32m',
  red: '\x1b[31m',
  yellow: '\x1b[33m',
  blue: '\x1b[34m',
  cyan: '\x1b[36m'
};

const log = {
  success: (msg) => console.log(`${colors.green}✓${colors.reset} ${msg}`),
  error: (msg) => console.log(`${colors.red}✗${colors.reset} ${msg}`),
  info: (msg) => console.log(`${colors.blue}ℹ${colors.reset} ${msg}`),
  warn: (msg) => console.log(`${colors.yellow}⚠${colors.reset} ${msg}`),
  section: (msg) => console.log(`\n${colors.cyan}▶${colors.reset} ${msg}`)
};

// Test data for all endpoints
const webhookTests = [
  // Workflow 1: Sell Process Started
  {
    workflow: 'Sell Process',
    endpoint: '/sell-started',
    body: {
      userId: 'user-test-' + Date.now(),
      timestamp: new Date().toISOString(),
      source: 'automated_test',
      userProfile: {
        displayName: 'Test User',
        email: 'test@globul.net',
        language: 'bg'
      }
    }
  },

  // Workflow 2: Vehicle Type Selected
  {
    workflow: 'Vehicle Type',
    endpoint: '/vehicle-type-selected',
    body: {
      userId: 'user-test-' + Date.now(),
      vehicleType: 'car',
      timestamp: new Date().toISOString(),
      sessionId: 'session-test-' + Date.now()
    }
  },

  // Workflow 3: Complete Sell Workflow
  {
    workflow: 'Complete Sell',
    endpoint: '/seller-type-selected',
    body: {
      userId: 'user-test-' + Date.now(),
      sellerType: 'private',
      timestamp: new Date().toISOString(),
      sessionId: 'session-test-' + Date.now()
    }
  },
  {
    workflow: 'Complete Sell',
    endpoint: '/vehicle-data-entered',
    body: {
      userId: 'user-test-' + Date.now(),
      sessionId: 'session-test-' + Date.now(),
      vehicleData: {
        make: 'BMW',
        model: 'X5',
        year: 2020,
        mileage: 45000,
        fuelType: 'diesel',
        transmission: 'automatic'
      },
      timestamp: new Date().toISOString()
    }
  },
  {
    workflow: 'Complete Sell',
    endpoint: '/equipment-selected',
    body: {
      userId: 'user-test-' + Date.now(),
      sessionId: 'session-test-' + Date.now(),
      equipment: ['air_conditioning', 'navigation', 'leather_seats'],
      timestamp: new Date().toISOString()
    }
  },
  {
    workflow: 'Complete Sell',
    endpoint: '/images-uploaded',
    body: {
      userId: 'user-test-' + Date.now(),
      sessionId: 'session-test-' + Date.now(),
      images: [
        { url: 'https://example.com/car1.jpg', type: 'exterior', order: 1 },
        { url: 'https://example.com/car2.jpg', type: 'interior', order: 2 }
      ],
      totalImages: 2,
      timestamp: new Date().toISOString()
    }
  },
  {
    workflow: 'Complete Sell',
    endpoint: '/price-set',
    body: {
      userId: 'user-test-' + Date.now(),
      sessionId: 'session-test-' + Date.now(),
      price: 35000,
      currency: 'EUR',
      negotiable: true,
      timestamp: new Date().toISOString()
    }
  },
  {
    workflow: 'Complete Sell',
    endpoint: '/contact-info-entered',
    body: {
      userId: 'user-test-' + Date.now(),
      sessionId: 'session-test-' + Date.now(),
      contactInfo: {
        name: 'Test User',
        phone: '+359888123456',
        email: 'test@globul.net',
        preferredContact: 'phone'
      },
      timestamp: new Date().toISOString()
    }
  },
  {
    workflow: 'Complete Sell',
    endpoint: '/listing-published',
    body: {
      userId: 'user-test-' + Date.now(),
      sessionId: 'session-test-' + Date.now(),
      listingId: 'listing-test-' + Date.now(),
      status: 'published',
      publishedAt: new Date().toISOString()
    }
  },

  // Workflow 4: User Tracking
  {
    workflow: 'User Tracking',
    endpoint: '/user-registered',
    body: {
      userId: 'user-test-' + Date.now(),
      email: 'test@globul.net',
      displayName: 'Test User',
      registrationMethod: 'email',
      timestamp: new Date().toISOString()
    }
  },
  {
    workflow: 'User Tracking',
    endpoint: '/user-logged-in',
    body: {
      userId: 'user-test-' + Date.now(),
      timestamp: new Date().toISOString(),
      loginMethod: 'email',
      deviceInfo: {
        type: 'desktop',
        os: 'Windows',
        browser: 'Chrome'
      }
    }
  },
  {
    workflow: 'User Tracking',
    endpoint: '/profile-updated',
    body: {
      userId: 'user-test-' + Date.now(),
      timestamp: new Date().toISOString(),
      updatedFields: ['phone', 'location']
    }
  },
  {
    workflow: 'User Tracking',
    endpoint: '/listing-created',
    body: {
      userId: 'user-test-' + Date.now(),
      listingId: 'listing-test-' + Date.now(),
      vehicleType: 'car',
      make: 'BMW',
      model: 'X5',
      year: 2020,
      price: 35000,
      timestamp: new Date().toISOString()
    }
  },

  // Workflow 5: User Engagement
  {
    workflow: 'User Engagement',
    endpoint: '/car-viewed',
    body: {
      userId: 'user-test-' + Date.now(),
      carId: 'car-test-' + Date.now(),
      timestamp: new Date().toISOString(),
      viewDuration: 45,
      source: 'search_results'
    }
  },
  {
    workflow: 'User Engagement',
    endpoint: '/favorite-added',
    body: {
      userId: 'user-test-' + Date.now(),
      carId: 'car-test-' + Date.now(),
      timestamp: new Date().toISOString(),
      source: 'listing_page'
    }
  },
  {
    workflow: 'User Engagement',
    endpoint: '/message-sent',
    body: {
      event: 'message_sent',
      messageId: 'msg-test-' + Date.now(),
      chatId: 'chat-test-' + Date.now(),
      carId: 'car-test-' + Date.now(),
      senderId: 'user-test-' + Date.now(),
      receiverId: 'seller-test-' + Date.now(),
      text: 'Test message from automated script',
      language: 'en',
      timestamp: new Date().toISOString()
    }
  },
  {
    workflow: 'User Engagement',
    endpoint: '/search-performed',
    body: {
      userId: 'user-test-' + Date.now(),
      timestamp: new Date().toISOString(),
      searchQuery: {
        make: 'BMW',
        yearFrom: 2018,
        yearTo: 2023
      },
      resultsCount: 15
    }
  },

  // Workflow 6: Admin & BI
  {
    workflow: 'Admin & BI',
    endpoint: '/admin-login',
    body: {
      action: 'admin_login',
      adminId: 'admin-test',
      timestamp: new Date().toISOString(),
      data: {
        isNewLocation: false,
        isNewDevice: false,
        multipleFailedAttempts: false
      }
    }
  },
  {
    workflow: 'Admin & BI',
    endpoint: '/admin-action',
    body: {
      action: 'admin_action',
      adminId: 'admin-test',
      timestamp: new Date().toISOString(),
      data: {
        operation: 'view_dashboard',
        targetId: 'dashboard-main'
      }
    }
  },
  {
    workflow: 'Admin & BI',
    endpoint: '/analytics-viewed',
    body: {
      action: 'analytics_viewed',
      adminId: 'admin-test',
      timestamp: new Date().toISOString(),
      data: {
        view: 'sales_dashboard',
        filters: { period: '7d' }
      }
    }
  },
  {
    workflow: 'Admin & BI',
    endpoint: '/system-alert',
    body: {
      action: 'system_alert',
      adminId: 'admin-test',
      timestamp: new Date().toISOString(),
      data: {
        severity: 'low',
        alertType: 'test_alert',
        message: 'Automated test alert'
      }
    }
  }
];

// Test a single webhook
async function testWebhook(test) {
  const url = BASE_URL + test.endpoint;
  
  try {
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(test.body)
    });

    const status = response.status;
    let result = { success: false, status, error: null };

    if (status >= 200 && status < 300) {
      result.success = true;
      log.success(`[${test.workflow}] ${test.endpoint} → ${status}`);
    } else {
      const errorText = await response.text();
      result.error = errorText;
      log.error(`[${test.workflow}] ${test.endpoint} → ${status}`);
      log.warn(`   Response: ${errorText.substring(0, 100)}`);
    }

    return result;
  } catch (error) {
    log.error(`[${test.workflow}] ${test.endpoint} → FAILED`);
    log.warn(`   Error: ${error.message}`);
    return { success: false, status: 0, error: error.message };
  }
}

// Run all tests
async function runAllTests() {
  console.log('\n╔═══════════════════════════════════════════════════════╗');
  console.log('║  Globul Cars - n8n Webhooks Automated Test Suite     ║');
  console.log('╚═══════════════════════════════════════════════════════╝\n');
  
  log.info(`Testing ${webhookTests.length} endpoints...`);
  log.info(`Base URL: ${BASE_URL}\n`);

  const results = [];
  let currentWorkflow = '';

  for (const test of webhookTests) {
    if (test.workflow !== currentWorkflow) {
      log.section(`Testing ${test.workflow}`);
      currentWorkflow = test.workflow;
    }

    const result = await testWebhook(test);
    results.push({ ...test, ...result });

    // Small delay between requests
    await new Promise(resolve => setTimeout(resolve, 500));
  }

  // Summary
  console.log('\n╔═══════════════════════════════════════════════════════╗');
  console.log('║  Test Summary                                         ║');
  console.log('╚═══════════════════════════════════════════════════════╝\n');

  const successful = results.filter(r => r.success).length;
  const failed = results.filter(r => !r.success).length;
  const total = results.length;
  const successRate = ((successful / total) * 100).toFixed(1);

  log.info(`Total Tests: ${total}`);
  log.success(`Successful: ${successful}`);
  if (failed > 0) {
    log.error(`Failed: ${failed}`);
  }
  log.info(`Success Rate: ${successRate}%`);

  if (failed > 0) {
    console.log('\n' + colors.yellow + 'Failed Endpoints:' + colors.reset);
    results.filter(r => !r.success).forEach(r => {
      console.log(`  - ${r.endpoint} (${r.status || 'Network Error'})`);
    });
  }

  console.log('\n' + colors.cyan + 'Next Steps:' + colors.reset);
  console.log('  1. Check n8n Executions: https://globul-cars-bg.app.n8n.cloud');
  console.log('  2. Verify workflows are Active');
  console.log('  3. Review any failed endpoints above\n');

  process.exit(failed > 0 ? 1 : 0);
}

// Run tests
runAllTests().catch(error => {
  log.error('Fatal error: ' + error.message);
  process.exit(1);
});
